samples/snippets/product/crud_product.py

```diff
@@ -0,0 +1,142 @@
+    return ProductServiceClient(client_options=client_options)
+
+
+# generate product for create
+def generate_product() -> Product:
+    price_info = PriceInfo()
+    price_info.price = 30.0
+    price_info.original_price = 35.5
+    price_info.currency_code = "USD"
+    return product.Product(
+        title='Nest Mini',
+        type_=product.Product.Type.PRIMARY,
+        categories=['Speakers and displays'],
+        brands=['Google'],
+        price_info=price_info,
+        availability="IN_STOCK",
+    )
+
+
+# generate product for update
+def generate_product_for_update() -> Product:
+    price_info = PriceInfo()
+    price_info.price = 20.0
+    price_info.original_price = 25.5
+    price_info.currency_code = "EUR"
+    return product.Product(
+        id=product_id,
+        name=product_name,
+        title='Updated Nest Mini',
+        type_=product.Product.Type.PRIMARY,
+        categories=['Updated Speakers and displays'],
+        brands=['Updated Google'],
+        availability="OUT_OF_STOCK",
+        price_info=price_info,
+    )
+
+
+# create product
+def create_product() -> object:
+    create_product_request = CreateProductRequest()
+    create_product_request.product = generate_product()
+    create_product_request.product_id = product_id
+    create_product_request.parent = default_branch_name
+
+    print("---create product request---")
+    print(create_product_request)
+
+    product_created = get_product_service_client().create_product(
+        create_product_request)
+    print("---created product:---")
+    print(product_created)
+    return product_created
+
+
+# get product
+def get_product() -> object:
+    get_product_request = GetProductRequest()
+    get_product_request.name = product_name
+
+    print("---get product request---")
+    print(get_product_request)
+
+    get_product_response = get_product_service_client().get_product(
+        get_product_request)
+
+    print("---get product response:---")
+    print(get_product_response)
+    return get_product_response
+
+
+# update product
+def update_product():
+    update_product_request = UpdateProductRequest()
+    update_product_request.product = generate_product_for_update()
+    update_product_request.allow_missing = True
+
+    print("---update product request---")
+    print(update_product_request)
+
+    updated_product = get_product_service_client().update_product(
+        update_product_request)
+    print('---updated product---:')
+    print(updated_product)
+    return updated_product
+
+
+# delete product
+def delete_product():
+    delete_product_request = DeleteProductRequest()
+    delete_product_request.name = product_name
+
+    print("---delete product request---")
+    print(delete_product_request)
+
+    get_product_service_client().delete_product(delete_product_request)
+
+    print("deleting product " + product_name)
+    print("---product was deleted:---")
+
+
+# call the methods
+create_product()
```

Comment:
Worrying that if anything goes wrong in between and the product is not deleted, the test will fail the next time CreateProduct is called. Should we always call delete_product() before this, or use a random ID instead?

Reply:
done